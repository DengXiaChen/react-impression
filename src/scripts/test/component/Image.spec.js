import React from 'react';
import { shallow } from 'enzyme';
import { Image } from '../../components/impression';

const setup = () => {
    const props = {
        fluid: true,
        rounded: true,
        circle: true,
        thumbnail: true,
    };

    const wrapper = shallow(<Image {...props} />);

    return {
        props,
        wrapper,
    };
};

describe('Button', () => {
    const { wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('img').exists()).toBe(true);
    });

    it('fluid', () => {
        expect(wrapper.find('.img-fluid').exists()).toBe(true);
    });

    it('rounded', () => {
        expect(wrapper.find('.img-rounded').exists()).toBe(true);
    });

    it('circle', () => {
        expect(wrapper.find('.img-circle').exists()).toBe(true);
    });

    it('thumbnail', () => {
        expect(wrapper.find('.img-thumbnail').exists()).toBe(true);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
