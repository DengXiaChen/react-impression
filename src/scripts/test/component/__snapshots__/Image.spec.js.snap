// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <img
    className="img-fluid img-rounded img-circle img-thumbnail"
/>,
  "nodes": Array [
    <img
      className="img-fluid img-rounded img-circle img-thumbnail"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Image
        circle={true}
        fluid={true}
        rounded={true}
        thumbnail={true}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Image {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "circle": true,
          "fluid": true,
          "rounded": true,
          "thumbnail": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <img
          className="img-fluid img-rounded img-circle img-thumbnail"
/>,
        "_debugID": 2,
        "_renderedOutput": <img
          className="img-fluid img-rounded img-circle img-thumbnail"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Image
    circle={true}
    fluid={true}
    rounded={true}
    thumbnail={true}
/>,
}
`;
